pkg/api/types.go

```diff
@@ -1202,6 +1208,21 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// LocalVolume represents a local volume accessible only from one node
+type LocalVolumeSource struct {
+	// NodeName represents the node that the volume is located at
+	NodeName string
+	// Fs represents a filesystem-based local volume
+	// +optional
+	Fs *LocalFsVolume
```

Comment:
> Is this a hypothetical scenario or do you have concrete examples?

Local volume proposal wants to cover both devices mounted outside of Kubernetes and block devices mounted by Kubernetes (or presented as raw devices to pods somehow). To me, these are two quite different volumes. The first one needs a path to local directory, the second one needs path to a device name (which is not a directory) and at least a `fsType` that should be created there if it's going to be used as regular PV (i.e. mounted by Kubernetes). Hence separate structures `FS` and `Block` (in future).

Overloading `Path` with both mount directory and device path looks really ugly to me.